File: src/weaponsprocurement/gui/WimGuiStyle.java
package weaponsprocurement.gui;

import java.awt.Color;

final class WimGuiStyle {
    static final float BUTTON_IDLE_DIM_MULT = 0.55f;
    static final float TEXT_TOP_PAD = 3f;
    static final float TEXT_LEFT_PAD = 8f;
    static final float TEXT_APPROX_CHAR_WIDTH = 6.8f;
    static final float TEXT_LINE_HEIGHT = 15f;
    static final float TEXT_VERTICAL_PADDING = 8f;

    static Color DEFAULT_TEXT = new Color(220, 220, 220);
    static Color WHITE_TEXT = Color.WHITE;
    static Color UNCOLOURED_BUTTON = new Color(0, 0, 0, 225);
    static Color DISABLED_BACKGROUND = new Color(28, 28, 28, 235);
    static Color DISABLED_DARK = new Color(18, 18, 18, 235);
    static Color DISABLED_TEXT = new Color(120, 120, 120);
    static Color DISABLED_BORDER = new Color(95, 95, 95);
    static Color ALERT_RED = new Color(245, 95, 85, 240);
    static Color SAVE_PURPLE = new Color(146, 126, 160, 225);
    static Color LOAD_YELLOW = new Color(188, 178, 86, 225);
    static Color CONFIRM_GREEN = new Color(144, 180, 148, 225);
    static Color CANCEL_RED = new Color(218, 142, 140, 225);
    static Color PRESET_SCOPE_ORANGE = new Color(202, 146, 112, 225);
    static Color SELECTED_BLUE = new Color(52, 78, 88, 225);
    static Color PANEL_HEADING = new Color(40, 40, 40, 225);
    static Color STALE_HEADING = new Color(120, 120, 120, 225);
    static Color COLLAPSIBLE_HEADING = new Color(80, 80, 80, 225);
    static Color UNSAVED_HEADING = new Color(220, 220, 220, 225);
    static Color TAG_CATEGORY_HEADING = new Color(210, 165, 185, 225);
    static Color MODAL_PANEL_BACKGROUND = new Color(0, 0, 0, 220);
    static Color MODAL_PANEL_BORDER = new Color(210, 210, 210, 220);
    static Color ROW_BORDER = new Color(210, 210, 210, 220);

    private WimGuiStyle() {
    }

    static Color dimForIdle(Color color) {
        if (color == null) {
            return DISABLED_DARK;
        }
        return new Color(
                clampColor(Math.round(color.getRed() * BUTTON_IDLE_DIM_MULT)),
                clampColor(Math.round(color.getGreen() * BUTTON_IDLE_DIM_MULT)),
                clampColor(Math.round(color.getBlue() * BUTTON_IDLE_DIM_MULT)),
                color.getAlpha());
    }

    private static int clampColor(int value) {
        return Math.max(0, Math.min(255, value));
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiStyle.kt
package weaponsprocurement.gui

import java.awt.Color

class WimGuiStyle private constructor() {
    companion object {
        const val BUTTON_IDLE_DIM_MULT = 0.55f
        const val TEXT_TOP_PAD = 3f
        const val TEXT_LEFT_PAD = 8f
        const val TEXT_APPROX_CHAR_WIDTH = 6.8f
        const val TEXT_LINE_HEIGHT = 15f
        const val TEXT_VERTICAL_PADDING = 8f

        @JvmField var DEFAULT_TEXT: Color = Color(220, 220, 220)
        @JvmField var WHITE_TEXT: Color = Color.WHITE
        @JvmField var UNCOLOURED_BUTTON: Color = Color(0, 0, 0, 225)
        @JvmField var DISABLED_BACKGROUND: Color = Color(28, 28, 28, 235)
        @JvmField var DISABLED_DARK: Color = Color(18, 18, 18, 235)
        @JvmField var DISABLED_TEXT: Color = Color(120, 120, 120)
        @JvmField var DISABLED_BORDER: Color = Color(95, 95, 95)
        @JvmField var ALERT_RED: Color = Color(245, 95, 85, 240)
        @JvmField var SAVE_PURPLE: Color = Color(146, 126, 160, 225)
        @JvmField var LOAD_YELLOW: Color = Color(188, 178, 86, 225)
        @JvmField var CONFIRM_GREEN: Color = Color(144, 180, 148, 225)
        @JvmField var CANCEL_RED: Color = Color(218, 142, 140, 225)
        @JvmField var PRESET_SCOPE_ORANGE: Color = Color(202, 146, 112, 225)
        @JvmField var SELECTED_BLUE: Color = Color(52, 78, 88, 225)
        @JvmField var PANEL_HEADING: Color = Color(40, 40, 40, 225)
        @JvmField var STALE_HEADING: Color = Color(120, 120, 120, 225)
        @JvmField var COLLAPSIBLE_HEADING: Color = Color(80, 80, 80, 225)
        @JvmField var UNSAVED_HEADING: Color = Color(220, 220, 220, 225)
        @JvmField var TAG_CATEGORY_HEADING: Color = Color(210, 165, 185, 225)
        @JvmField var MODAL_PANEL_BACKGROUND: Color = Color(0, 0, 0, 220)
        @JvmField var MODAL_PANEL_BORDER: Color = Color(210, 210, 210, 220)
        @JvmField var ROW_BORDER: Color = Color(210, 210, 210, 220)

        @JvmStatic
        fun dimForIdle(color: Color?): Color {
            if (color == null) {
                return DISABLED_DARK
            }
            return Color(
                clampColor(Math.round(color.red * BUTTON_IDLE_DIM_MULT)),
                clampColor(Math.round(color.green * BUTTON_IDLE_DIM_MULT)),
                clampColor(Math.round(color.blue * BUTTON_IDLE_DIM_MULT)),
                color.alpha,
            )
        }

        private fun clampColor(value: Int): Int = Math.max(0, Math.min(255, value))
    }
}
